bridges/snowbridge/primitives/core/src/location.rs

```diff
@@ -23,23 +23,36 @@ pub type AgentId = H256;
 
 /// Creates an AgentId from a Location. An AgentId is a unique mapping to an Agent contract on
 /// Ethereum which acts as the sovereign account for the Location.
-/// Resolves Polkadot locations (as seen by Ethereum) to unique `AgentId` identifiers.
-pub type AgentIdOf = HashedDescription<
-	AgentId,
-	(
-		DescribeHere,
-		DescribeFamily<DescribeAllTerminal>,
-		DescribeGlobalPrefix<(DescribeTerminus, DescribeFamily<DescribeTokenTerminal>)>,
-	),
->;
+#[deprecated(note = "Use LocationHashOf instead.")]
+pub type AgentIdOf =
+	HashedDescription<AgentId, (DescribeHere, DescribeFamily<DescribeAllTerminal>)>;
 
 pub type TokenId = H256;
 
 /// Convert a token location (relative to Ethereum) to a stable ID that can be used on the Ethereum
 /// side
+#[deprecated(note = "Use LocationHashOf instead.")]
 pub type TokenIdOf = HashedDescription<
 	TokenId,
-	DescribeGlobalPrefix<(DescribeTerminus, DescribeFamily<DescribeTokenTerminal>)>,
+	DescribeForEthereum<(DescribeTerminus, DescribeFamily<DescribeSnowbridgeTerminal>)>,
+>;
+
+/// Resolves Polkadot locations (as seen by Ethereum) to unique `H256` identifiers.
+/// All V1 converters are inherited here for backwards compatibility
+/// With AliasOrigin enforced in V2 the user origin from source chain will be reserved and sent to
+/// Ethereum.
+/// For asset transfers, the legacy AH agent ID is hardcoded in the V2 contract side,
+/// which ignores the user origin, so there won't be any compatibility issue.
+/// For transact calls, the user origin will be respected in the V2 contract side, but won't
+/// conflict with the legacy AH.
+pub type LocationHashOf = HashedDescription<
+	H256,
+	(
+		DescribeHere,
+		DescribeFamily<DescribeAllTerminal>,
+		DescribeForEthereum<(DescribeTerminus, DescribeFamily<DescribeSnowbridgeTerminal>)>,
+		DescribeForEthereum<(DescribeTerminus, DescribeFamily<DescribeAllTerminal>)>,
```

Comment:
Please add comments explaining each case in the tuple, and when it would or would not be triggered.

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/28/commits/dcb8d7b419a0294b3c9a4e7cc5b84c2fb97595a3